docs/proposals/002-Versioning_Proposal.md

```diff
@@ -0,0 +1,209 @@
+# Proposal : Tag-Based Version Management System for DISCOVER Cookbook
+
+## Abstract
+
+This proposal outlines an implementation strategy for a tag-based version management system for the DISCOVER Cookbook following standard documentation versioning practices. The implementation will enable users to seamlessly navigate between different cookbook editions through a simple version selector integrated with the existing Jupyter Book interface, while maintaining the current GitHub Pages infrastructure.
+
+## Detailed description
+
+### Current Architecture Analysis
+
+The DISCOVER Cookbook currently has no formal versioning system:
+
+- Current version is in the main branch, deployed at https://discover-cookbook.numfocus.org
+
+This structure creates several critical challenges:
+
+1. **Hidden Versions**: Users can't tell which version they're viewing or that other versions exist, so they might miss updated recommendations.
+
+2. **Hard to Navigate**: Switching between versions requires typing different URLs or navigating through GitHub, making for a poor user experience.
+
+3. **Content Evolution Opacity**: Users can't easily see how DEI recommendations have changed over time.
+
+
+### Solution Architecture
+
+My implementation will use a tag-based versioning approach :
+
+1. **Tag-Based Version Management:**:
+   - Maintain a single active branch (main)
+   - Use Git tags to mark specific versions (latest version , previous versions)
```

Comment:
we should stick to numeric versions. 1.0, 2.0, and then the latest version will just be the main branch and can be labelled 2.0-dev. One might even look at how to included [versioneer](https://github.com/python-versioneer/python-versioneer) so you can see which version of the book is being renderd.